Sources/SymbolKit/SymbolGraph/SymbolGraph+Overloads.swift

```diff
@@ -107,6 +104,58 @@ extension SymbolGraph {
     }
 }
 
+protocol OverloadsSortable {
+    var declaration: String? { get }
+    var identifierKey: String { get }
+}
+
+extension OverloadsSortable {
+    static func sortOverloadsByDeclaration(_ lhs: Self, _ rhs: Self) -> Bool {
+        guard let lhsDeclaration = lhs.declaration, let rhsDeclaration = rhs.declaration else {
+            preconditionFailure("Attempting to sort overloads by declaration, but one of the overloads did not have a declaration. lhs: '\(lhs.identifierKey)' ('\(lhs.declaration ?? "")') rhs: '\(rhs.identifierKey)' ('\(rhs.declaration ?? "")')")
```

Comment:
Thanks for typing this message out - will save someone's time in the future I'm sure!